### PR TITLE
feat: Add new ptn modules `avm/ptn/azd/aks`.

### LINKED ISSUE #1224: [Module Proposal]: `avm/ptn/azd/aks`
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes.

### Bicep or Terraform?

Bicep

### Module Classification?

Pattern Module

### Module Name

avm/ptn/azd/aks

### Module Details

Please add a Pattern Module named `avm/ptn/azd/aks`, this module will create the following three resources:

1. Azure Kubernetes Service (AKS) Managed Cluster.
2. Azure Container Registry (ACR).
3. Key Vault.

For details, please refer to azd: 
https://github.com/Azure/azure-dev/blob/main/templates/common/infra/bicep/core/host/aks.bicep

CC: @jongio

### Do you want to be the owner of this module?

No

### Module Owner's GitHub Username (handle)

jongio

### (Optional) Secondary Module Owner's GitHub Username (handle)

_No response_

### PR DESCRIPTION
**Pull request by jerryshia**
## Description
Fixes https://github.com/Azure/Azure-Verified-Modules/issues/1224.

## Pipeline Reference

| Pipeline |
| -------- |
|  [avm.ptn.azd.aks](https://github.com/jerryshia/bicep-registry-modules/actions/workflows/avm.ptn.azd.aks.yml)  |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

@jongio for notification.

### CODE REVIEW

File: .github/workflows/avm.ptn.azd.aks.yml
Comment (AlexanderSehr) [inline .github/workflows/avm.ptn.azd.aks.yml:31]: outdated

File: avm/ptn/azd/aks/main.bicep
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/main.bicep:95]: Must be changed to 
```suggestion
param monitoringWorkspaceResourceId string?
```

if it's actually the resource and not workspaceID. This needs to be fixed in other places too

File: avm/ptn/azd/aks/tests/e2e/defaults/main.test.bicep
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/tests/e2e/defaults/main.test.bicep:72]: Many of the below parameters don't seem to be mandatory and should hence not be part of the test that wants to explicitely test only the minimum parameter set. Was this intional?

File: avm/ptn/azd/aks/tests/e2e/max/main.test.bicep
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/tests/e2e/max/main.test.bicep:4]: Incorrect description for the type of test

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
After the suggested changes please ensure to
- Re-run the [Toggle workflows](https://github.com/jerryshia/bicep-registry-modules/actions/workflows/platform.toggle-avm-workflows.yml) workflow to disable all your workflows, and only enable the one you want to test (like the one for this pattern). This will ensure the trigger on main does not start running all workflows when you sync your fork, etc.
- Resolve the conflicts
- Re-run the `Set-AVMModule` script to update the generated files
- Re-run the pipeline end-2-end to validate all tests pass

Thanks @jerryshia . Great work 💪

**Comment by ReneHezser** [comment]
@jerryshia 
We have added examples for Bicep parameter files to the Readme. This has been applied to all published modules but needs to be done for PRs as well. Can you please run the Set-AVMModule utility as detailed [here](https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/generate-bicep-module-files/). It is required for the validation pipeline to succeed and the contribution to be published.

Please reach out if any support is needed.

**Comment by jerryshia** [comment]
@AlexanderSehr I have updated based on your comments, please re-review.

File: avm/ptn/azd/aks/tests/e2e/defaults/main.test.bicep
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/tests/e2e/defaults/main.test.bicep:18]:
```suggestion
param serviceShort string = 'paamin'
```

Comment (AlexanderSehr) [reply avm/ptn/azd/aks/tests/e2e/defaults/main.test.bicep:18]: Requires update of readme

File: avm/ptn/azd/aks/tests/e2e/max/main.test.bicep
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/tests/e2e/max/main.test.bicep:18]:
```suggestion
param serviceShort string = 'paamax'
```

Comment (AlexanderSehr) [reply avm/ptn/azd/aks/tests/e2e/max/main.test.bicep:18]: Requires update to readme

File: avm/ptn/azd/aks/main.bicep
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/main.bicep:137]: Redundant?

File: avm/ptn/azd/aks/main.bicep
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/main.bicep:345]: the description says it's required, but the `?` makes it optional. Which is it? :wink:

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr I have updated based on your comments, please re-review.

Much appreciated @jerryshia . I had a few more findings, but all in all you're almost done 💪 Let me know if you have any questions

File: avm/ptn/azd/aks/main.bicep
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/main.bicep:194]: Misses passthru of `enableTelemetry` parameter

File: avm/ptn/azd/aks/main.bicep
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/main.bicep:261]: Misses passthru of `enableTelemetry` parameter

File: avm/ptn/azd/aks/main.bicep
Comment (AlexanderSehr) [reply avm/ptn/azd/aks/main.bicep:192]: Not the correct fix. Please don't add the telemetry parameter as a condition, but just pass the parameter in the params block through to the linked module. Also please don't self-resolve comments 😉

File: avm/ptn/azd/aks/main.bicep
Comment (AlexanderSehr) [reply avm/ptn/azd/aks/main.bicep:259]: Not the correct fix. Please don't add the telemetry parameter as a condition, but just pass the parameter in the params block through to the linked module. Also please don't self-resolve comments 😉

File: avm/ptn/azd/aks/main.bicep
Comment (AlexanderSehr) [inline avm/ptn/azd/aks/main.bicep:297]: Not the correct fix. Please don't add the telemetry parameter as a condition, but just pass the parameter in the params block through to the linked module.

File: .github/CODEOWNERS
Comment (jerryshia) [reply .github/CODEOWNERS:16]: The latest commit is about aks, please check

File: avm/ptn/azd/aks/main.bicep
Comment (jerryshia) [reply avm/ptn/azd/aks/main.bicep:192]: I have updated by your comments.

File: avm/ptn/azd/aks/main.bicep
Comment (jerryshia) [reply avm/ptn/azd/aks/main.bicep:260]: I have updated by your comments.

File: avm/ptn/azd/aks/main.bicep
Comment (jerryshia) [reply avm/ptn/azd/aks/main.bicep]: I have updated by your comments.

File: .github/CODEOWNERS
Comment (AlexanderSehr) [reply .github/CODEOWNERS:16]: Looks good, thanks. Now we just have to ensure that (as per the [specs](https://azure.github.io/Azure-Verified-Modules/specs/shared/#team-memberships)) the team is created and added to the correct parent team. This will enable you to subsequently review & approve PRs on your own (unless you're the author, of course) 😄

File: avm/ptn/azd/aks/main.bicep
Comment (eriqua) [inline avm/ptn/azd/aks/main.bicep:1]: This will be used as the readme title. I'd suggest a more user friendly short name, possibly starting with `Azd`, like e.g.
```suggestion
metadata name = 'Azd AKS'
```

Please note that the readme would require an update after that

File: .github/CODEOWNERS
Comment (AlexanderSehr) [reply .github/CODEOWNERS:16]: Seems alright now.

File: avm/ptn/azd/aks/main.bicep
Comment (jerryshia) [reply avm/ptn/azd/aks/main.bicep]: I have updated by your comments.

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Looks good to me @jerryshia. One last thing before we can merge: Please resolve the conflicts 😉 We're merging quite a few AZD modules in these days and I think they're all causing each other conflicts in the CODEOWNERS & issue template files 😄

**Comment by jerryshia** [comment]
@AlexanderSehr The conflict has been resolved.

File: .github/ISSUE_TEMPLATE/avm_module_issue.yml
Comment (AlexanderSehr) [inline .github/ISSUE_TEMPLATE/avm_module_issue.yml:51]: Thanks @jerryshia, the PR is good to go. Something went wrong with the resolution of conflicts for the ML one though, we we have to wait until this PR is merged (and may need to resolve conflicts again): https://github.com/Azure/bicep-registry-modules/pull/3511
Otherwise, the static tests in upstream will prevent this module from being published. Will let you know once it is merged.

Comment (AlexanderSehr) [reply .github/ISSUE_TEMPLATE/avm_module_issue.yml:51]: Ok, no conflict in sight. Approving

**Review by AlexanderSehr — APPROVED** [review]
🚀

**Pull request merged**
